edb/schema/objects.py

```diff
@@ -786,6 +852,27 @@ class Object(s_abc.Object, s_abc.ObjectContainer, metaclass=ObjectMeta):
 
     _fields: Dict[str, SchemaField[Any]]
 
+    def schema_reduce(self) -> Tuple[str, uuid.UUID]:
+        return type(self).__name__, self.id
+
+    @classmethod
+    @functools.lru_cache(maxsize=10240)
+    # mypy hates lru_cache
+    def schema_restore(  # type: ignore
```

Comment:
I'm curious, will mypy ignore this function here but still keep validating that it's being called correctly from other places?

Reply:
Mypy complains about the signature not matching that of the parent, because `@lru_cache` adds some properties to the returned callable (cache information).  This has no effect on the actual function signature.

Reply:
Though using @lru_cache does probably lose the attribute types as far as mypy is concerned